### PR TITLE
[Rails 5] Payment method params

### LINKED ISSUE #7027: [Rails 5.0] transaction fees not persistent
## Description

Values inserted on the "Calculator" section of a payment method do not persist, after clicking "Update"

## Expected Behavior

Changes on transaction fees should be persistent.

## Actual Behaviour

They are not; this prevents setting a transaction fee, to a given payment method

## Steps to Reproduce

1. Go to admin/payment_methods/<payment_method_id>/edit
2. Select a calculator and fill in some values
3. Click "Update"
4. See how the changes are gone

## Animated Gif/Screenshot

Peek 2021-03-05 20-50

## Possible Fix

Related to strong params?

Similar to #5568 and the fix #5570. Probably good to add test cases on the `payment_methods_controller_spec.rb`. After spotting this, it becomes clear that the "Calculator" section is not well covered by feature specs... reopened #6939. Although we might not need both controller and feature specs here. Maybe something to chat about.

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Closes #7027

Fixes some issues with params in `Admin::PaymentMethodsController` and adds test coverage.

#### What should we test?

See #7027.

### CODE REVIEW

File: app/controllers/spree/admin/payment_methods_controller.rb [hunk 1]
```diff
@@ -34,16 +32,16 @@ def update
         force_environment
 
         invoke_callbacks(:update, :before)
-        payment_method_type = params[:payment_method].delete(:type)
-        if @payment_method['type'].to_s != payment_method_type
+
+        if @payment_method['type'].to_s != payment_method_class
           @payment_method.update_columns(
-            type: payment_method_type,
+            type: payment_method_class,
```
Comment (sauloperez) [inline app/controllers/spree/admin/payment_methods_controller.rb:38]: given the result of the earlier call isn't memorized, isn't this `nil`? The second time `base_params` no longer contains the `:type` key.

Comment (Matt-Yorkley) [reply app/controllers/spree/admin/payment_methods_controller.rb:38]: Woah, nice catch.

Comment (Matt-Yorkley) [reply app/controllers/spree/admin/payment_methods_controller.rb:38]: Fixed, and test coverage added :+1:

**Review by sauloperez — APPROVED** [review]

**Comment by filipefurtad0** [comment]
Ok, fixed :ok_hand: 

Tested the scenario around #7027:
- editing a payment method -> setting a transaction fee
- placing an order in the shopfront -> the set enterprise fee appears at /checkout
- placed the order -> transaction fee appears on order total and confirmation
- and also on /orders and /reports


Checked whether the Payments reports is now working, after this PR - #7021:

- but unfortunately not: we're still getting the snail for orders placed by the rails5-hub
- Checking other reports, from other hubs (as superadmin) works as before 

Good to go :+1:
(and keeping #7021 open)

**Comment by Matt-Yorkley** [comment]
In relation to #7021, I think some of the orders we've created on UK Staging while this bug was active are in a broken state. The fix should stop us creating more like that, but it won't magically fix the broken orders that were created already. I suspect if we edit and hit "save and recalculate" on those orders (to make the payment and shipment states show up again) then the report will be working again...

**Comment by Matt-Yorkley** [comment]
Thanks @filipefurtad0 :smile:

**Pull request merged**